csharp/test/Apache.Arrow.Tests/TestData.cs

```diff
@@ -21,12 +21,13 @@ namespace Apache.Arrow.Tests
 {
     public static class TestData
     {
-        public static RecordBatch CreateSampleRecordBatch(int length)
+        //TODO: Remove the createDictionaryArray argument after all writer/reader supports DictionaryType serialization
+        public static RecordBatch CreateSampleRecordBatch(int length, bool createDictionaryArray = false)
```

Comment:
A general comment about `createDictionaryArray` - I think the existing tests should be left alone. That way we are still testing as much as we can without any dictionaries. Then we create new tests for dictionary support.

An easy way to do this is to change a bunch of tests from `[Fact]` to `[Theory]` and pass in `true` and `false` into the test for `bool createDictoinaryArray`. That way the test runs twice, once with a dictionary, and once without. That should give us the best test coverage.

Reply:
I understand.
I fixed tests to use `[Theory]`